app/assets/javascripts/oxalis/view/version_view.js

```diff
@@ -17,24 +17,39 @@ export type Versions = {
   volume?: ?number,
 };
 
-type Props = {
+type StateProps = {
   tracing: Tracing,
 };
 
+type Props = StateProps & {
+  allowUpdate: boolean,
+};
+
 type State = {
   activeTracingType: "skeleton" | "volume",
+  originalAllowUpdate: boolean,
 };
 
 class VersionView extends React.Component<Props, State> {
   state = {
     activeTracingType: this.props.tracing.skeleton != null ? "skeleton" : "volume",
+    originalAllowUpdate: false,
   };
 
+  componentDidMount() {
+    // Remember whether the tracing could originally be updated
+    this.setState({ originalAllowUpdate: this.props.allowUpdate });
```

Comment:
Why does this prop need to be written into state? If I'm not missing anything, the state is never changed? Couldn't we access the prop directly where needed? If we need to keep it, I'd rename it to "initialAllowUpdate" btw, since this is more idiomatic afaik :)

Reply:
The problem is that we change the `allowUpdate` value while the version restore view is open and therefore the prop changes as well. We need the value `allowUpdate` has when the component is mounted. I'll rename it to `initialAllowUpdate`.

Reply:
Hm, ok, but where is `allowUpdate` set to true? I only see the calls to `setAnnotationAllowUpdateAction`, where the initial value is passed (which would be false in the case we are talking about).

Reply:
Not sure whether we're talking about the same thing :D
If we're in a read-only tracing, `allowUpdate` is false when opening the version view, therefore, `initialAllowUpdate` is false as well and when closing the view `setAnnotationAllowUpdateAction` will set `allowUpdate` to `initialAllowUpdate` which is false - unnecessary in that case, but ok.
If we're in a normal tracing, ' allowUpdate' is true when opening the version view, `initialAllowUpdate` as well. We'll then set `allowUpdate` to false, because we don't want the tracing to be modified while the version is open. When closing the view or restoring an older version `setAnnotationAllowUpdateAction` will set `allowUpdate` to `initialAllowUpdate` which is true, so the user can modify the tracing again.
Hope that clears things up :)

Reply:
Ahh, ok, sorry, you had to type this out :see_no_evil: